packages/@ember/-internals/runtime/lib/mixins/array.d.ts

```diff
@@ -0,0 +1,103 @@
+type Value<T, K extends string> = K extends keyof T ? T[K] : unknown;
+
+export type NatifiedArray<T> = T[] & NativeArray<T>;
+
+interface EmberArray<T> {
+  length: number;
+  objectAt(idx: number): T | undefined;
+}
+declare class EmberArray<T> {
```

Comment:
Why the separate interface merged with the class?

Reply:
It's probably wrong. The `interface` was supposed to be stuff you have to declare. The way Mixins are handled always confuses me!